common/src/app/common/types/pages_list.cljc

```diff
@@ -6,22 +6,27 @@
 
 (ns app.common.types.pages-list
   (:require
-   [app.common.data :as d]))
+   [app.common.data :as d]
+   [app.common.pages.helpers :as cph]))
 
 (defn get-page
   [file-data id]
   (get-in file-data [:pages-index id]))
 
 (defn add-page
   [file-data page]
-  (let [; It's legitimate to add a page that is already there,
+  (let [page-num (:page-num page)
+        page (dissoc page :page-num)
+
+        ; It's legitimate to add a page that is already there,
         ; for example in an idempotent changes operation.
-        conj-if-not-exists (fn [pages id]
-                             (cond-> pages
-                               (not (d/seek #(= % id) pages))
-                               (conj id)))]
+        add-if-not-exists (fn [pages id]
+                            (when (not (d/seek #(= % id) pages))
```

Comment:
the `when` returns nil if condition evaluates to false, I think this is not expected because we should not assoc nil to pages on file-data

Reply:
fixed